src/test/java/com/example/bookstore/controller/ShoppingCartControllerTest.java

```diff
@@ -0,0 +1,146 @@
+package com.example.bookstore.controller;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers.springSecurity;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.example.bookstore.dto.cart.items.CartItemDto;
+import com.example.bookstore.dto.cart.items.CreateCartItemRequestDto;
+import com.example.bookstore.dto.shopping.cart.CartItemUpdateRequestDto;
+import com.example.bookstore.dto.shopping.cart.ShoppingCartDto;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import java.util.Collections;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class ShoppingCartControllerTest {
+    protected static MockMvc mockMvc;
+    private static ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void beforeAll(@Autowired WebApplicationContext applicationContext) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+        objectMapper = new ObjectMapper().registerModule(new JavaTimeModule());
+    }
+
+    @Test
+    @DisplayName("Get shopping cart by authenticated user")
+    @WithMockUser(username = "test@gmail.com", authorities = {"USER"})
+    @Sql(scripts = {"classpath:database/cart-items/delete-all-cart-items.sql",
+            "classpath:database/shopping-cart/delete-all-carts.sql",
+            "classpath:database/users/delete-all-users.sql",
```

Comment:
I can see that some scripts are duplicated between tests

you can place these common scripts on class level


also remember that you'll need to merge method-level scripts and class-level scripts

also think of what execution phase should be selected for what scripts